### DISCUSSION TITLE
fixation cross in go nogo task

### DISCUSSION THREAD

**Discussion by mathieudev88** [jspsych/jsPsych#1873] [Q&A]
Hello,

Im starting in jsPsych, I would like to add a fixation cross in this go nogo task (task from experimental factory), see in the attached file.

I create the variable "fixation_block" in the script but I don't know how can I add the cross just before each stimulus in this script :

```
var fixation_block = {
  type: 'poldrack-text',
  text: '*',
  timing_stim: 500,
  timing_response: 500,
  timing_post_trial: 250,
  data: {
    trial_id: "fixation"
  },

};
```
I have tried with this line "test_stimuli_block.push(fixation_block)" just before the line "test_stimuli_block.push({" but no success :

```
var stims = jsPsych.randomization.shuffle([["orange", "stim1"],["blue","stim2"]])
var gap = 0
var current_trial = 0
var practice_stimuli = [{
  stimulus: '<div class = centerbox><div  id = ' + stims[0][1] + '></div></div>',
  data: {
    correct_response: correct_responses[0][1],
    condition: correct_responses[0][0],
    trial_id: 'practice'
  },
  key_answer: correct_responses[0][1]
}, {
  stimulus: '<div class = centerbox><div id = ' + stims[1][1] + '></div></div>',
  data: {
    correct_response: correct_responses[1][1],
    condition: correct_responses[1][0],
    trial_id: 'practice'
  },
  key_answer: correct_responses[1][1]
}];

//set up block stim. test_stim_responses indexed by [block][stim][type]
var test_stimuli_block = [{
  stimulus: '<div class = centerbox><div id = ' + stims[1][1] + '></div></div>',
  data: {
    correct_response: correct_responses[1][1],
    condition: correct_responses[1][0],
    trial_id: 'test_block'
  }
}];

for (var i = 0; i < num_go_stim; i++) {
  test_stimuli_block.push(fixation_block)
  test_stimuli_block.push({
    stimulus: '<div class = centerbox><div  id = ' + stims[0][1] + '></div></div>',
    data: {
      correct_response: correct_responses[0][1],
      condition: correct_responses[0][0],
      trial_id: 'test_block'
    }
  })
}

var practice_trials = jsPsych.randomization.repeat(practice_stimuli, 5); 
var test_trials = jsPsych.randomization.repeat(test_stimuli_block, 35); 
``` 

And the test_block variable  :

```
/* define test block */
var test_block = {
  type: 'poldrack-single-stim',
  timeline: test_trials,
  data: {
    trial_id: "stim",
    exp_stage: "test"
  },
  is_html: true,
  choices: [32],
  timing_stim: 750,
  timing_response: get_response_time,
  timing_post_trial: 0,
  on_finish: appendData
};

```

Do you have any ideas how can I implement the fixation cross in this script ?

Thanks in advance

[go-nogo-fixation cross.zip](https://github.com/jspsych/jsPsych/files/6591307/go-nogo-fixation.cross.zip)
Mat

**Comment by jodeleeuw** [comment]
Hey @mathieudev88. Can you please copy the relevant bits of code into the post so folks don't have to download a zip file?

**Reply by mathieudev88** [reply]
yes sorry, I have added the code, thanks.

**Reply by jodeleeuw** [reply]
Thanks! What happens with your current implementation of adding the `push()` statement? You mentioned that it wasn't successful, but what exactly happened?

**Reply by mathieudev88** [reply]
Yes sorry I did not specify. When Im adding the push(fixation_block) statement, the task works well but the fixation cross is not displayed before the stimuli.

**Comment by jodeleeuw** [comment]
I think the problem is that this for loop creates a set of test stimuli...

```js
for (var i = 0; i < num_go_stim; i++) {
  test_stimuli_block.push(fixation_block)
  test_stimuli_block.push({
    stimulus: '<div class = centerbox><div  id = ' + stims[0][1] + '></div></div>',
    data: {
      correct_response: correct_responses[0][1],
      condition: correct_responses[0][0],
      trial_id: 'test_block'
    }
  })
}
```

... which are then randomly repeated with this line of code:

```js
var test_trials = jsPsych.randomization.repeat(test_stimuli_block, 35); 
```

So `test_stimuli` is never used directly as input to the timeline as far as I can tell.

You need to instead insert the fixation cross in the part of the experiment that is using `test_trials`, but I don't think you've got that piece of code included in the snippets yet.

**Reply by mathieudev88** [reply]
Yes, this is effectively the problem with this loop to add the cross. Following your indication I’m looking to fix the problem by adding the fixation cross in  'test_trials'. Thanks

**Reply by mathieudev88** [reply]
Hey, @jodeleeuw I rework this code because the problem with this fixation cross is still without a solution for me, sorry. I can't add the fixation cross in the part of the experiment using test_trials. Test_trials is just used in the test_block variable, and for this line "var test_trials = jsPsych.randomization.repeat(test_stimuli_block, 35);"

I understand the problem with the loop, but I don't know how to overpass this. I have tried to make a timeline and add the fixation cross with this code but the cross just appear on the first trial (not on all trials) :
 
`var timeline_trials = {
  timeline: [fixation_block, test_block],
}

/* create experiment definition array */
var experiment_1 = [];
experiment_1.push(instruction_node);
experiment_1.push(attention_node)
experiment_1.push(reset_block)
experiment_1.push(start_test_block);
experiment_1.push(timeline_trials);
experiment_1.push(attention_node)
experiment_1.push(end_block)`
